exercise/reveal_models.py

```diff
@@ -54,24 +55,21 @@ def is_revealed(self, state: 'BaseRevealState', time: Optional[datetime.datetime
             RevealRule.TRIGGER.DEADLINE,
             RevealRule.TRIGGER.DEADLINE_ALL
         ]:
-            if time is None:
-                time = timezone.now()
-            reveal_time = self.get_reveal_time(state)
-            if reveal_time is not None:
-                return time > reveal_time
+            if self._is_revealed_due_to_time(state, time):
+                return True
         elif self.trigger == RevealRule.TRIGGER.COMPLETION:
-            points = state.get_points()
-            max_points = state.get_max_points()
-            if points is not None and max_points is not None:
-                if points >= max_points:
-                    return True
+            if self._is_revealed_due_to_full_points(state):
+                return True
             submissions = state.get_submissions()
             max_submissions = state.get_max_submissions()
             if submissions is not None and max_submissions is not None:
                 if max_submissions == 0:
                     return False
                 if submissions >= max_submissions:
                     return True
+        elif self.trigger == RevealRule.TRIGGER.DEADLINE_OR_FULL_POINTS:
+            if self._is_revealed_due_to_full_points(state) or self._is_revealed_due_to_time(state, time):
+                return True
```

Comment:
You can shorten this to ``return self._is_revealed_due_to_full_points(state) or self._is_revealed_due_to_time(state, time)``